### PR TITLE
Blog App: validations, model specs, and n+1 problems

### PR DESCRIPTION
**Pull request by mmatongo**
### Project requirements
- Add the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.
- Add flash messages in the create actions in all your controllers.
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user. This is the corresponding wireframe view: blog all user posts

### CODE REVIEW

File: spec/requests/comments_spec.rb [hunk 1]
```diff
@@ -1,7 +0,0 @@
-require 'rails_helper'
-
-RSpec.describe 'Comments', type: :request do
-  describe 'GET /index' do
-    pending "add some examples (or delete) #{__FILE__}"
-  end
```
Comment (SajjadAhmad14) [inline spec/requests/comments_spec.rb:6]: Please add unit tests for `Comment` model

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -1,5 +1,78 @@
 require 'rails_helper'
 
 RSpec.describe Post, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (SajjadAhmad14) [inline spec/models/post_spec.rb:4]: Please add unit test for `#liked?` of Post model.

File: app/models/post.rb [hunk 3]
```diff
@@ -1,4 +1,7 @@
 class Post < ApplicationRecord
+  validates :title, presence: true, length: { maximum: 250, too_long: '%<count>s characters is the maximum allowed' }
+  validates :comments_counter, :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```
Comment (SajjadAhmad14) [inline app/models/post.rb:3]: CommentsCounter validation should check if it's an integer and it's not equal to but greater than zero.
LikesCounter validation should check if it's an integer and it's not equal to but greater than zero.

File: app/models/user.rb [hunk 4]
```diff
@@ -1,9 +1,12 @@
 class User < ApplicationRecord
+  validates :name, presence: true
+  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```
Comment (SajjadAhmad14) [inline app/models/user.rb:3]: PostsCounter validation should check if it's an integer and it's not equal to but greater than zero.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Daniel, 
 ### You have been doing a tremendous job so far :ok_hand:
## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
README is professional. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/models/user.rb (hunk 4 above)
Comment (mmatongo) [reply app/models/user.rb:3]: How then do I create a post with no likes or comments?

File: app/models/post.rb (hunk 3 above)
Comment (mmatongo) [reply app/models/post.rb:3]: How then do I create a post with no likes or comments?

How do I create a user with likes and comments from signup

File: app/models/user.rb (hunk 4 above)
Comment (mmatongo) [reply app/models/user.rb:3]: How do I create a user with likes and comments from signup

**Comment by mmatongo** [comment]
> ## Required Changes ♻️
> > "Change is inevitable. Change is constant. -Benjamin Disraeli"
> 
> ## Hi Daniel,
> ### You have been doing a tremendous job so far 👌
> ## To Highlight 🎉
> PR has a title and summary. ✔️ README is professional. ✔️
> 
> ## To Improve 🙏
> * Please see this error which occurs when the user clicks on the like button on a post.
>   likes
> * [ ]  Please check inline comments under the review
> 


This error probably came about as a result of the database's being dirty, like functionality works just fine for me here.

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉🎉 

## Hi @mmatongo, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mmatongo** [comment]
> # Congratulations🎉🎉 
> 
> 
> 
> ## Hi @mmatongo, 
> 
> 
> 
> Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
> 
> Congratulations! 🎉 
> 



Thanks

**Pull request merged**